### PR TITLE
OS X support, optional single cmake compile

### PR DESCRIPTION
**Pull request by ahundt**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,55 +1,103 @@
...
+        alias ls='ls -G'
+        ;;
+      'WindowsNT')
+        OS='Windows'
+        ;;
+      'Darwin') 
+        OS='Mac'
+        /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+        brew tap homebrew/science
+        brew tap ahundt/robotics
```
Comment (mp3guy) [inline build.sh]: I can't merge something that hooks into your own third party repo into master. If you must, just fork ElasticFusion and do it on your own fork.

Comment (ahundt) [inline build.sh]: It just includes a pangolin install script. I can just make a branch that takes that part out, would it then be ok?

**Comment by stephane-lb** [comment]
Awesome news, @ahundt good job! You did not open the issue in your github repo, so there is a small issue there : 
```
./build.sh: line 78: syntax error near unexpected token `else'
./build.sh: line 78: `else'
```

**Comment by stephane-lb** [comment]
And you might have create a CMakeList.txt at the root of the repo that you forgot to commit too otherwise  the line 86 is a non sense :-)

**Comment by ahundt** [comment]
@stephane-lb I did create a cmakelists.txt at the root, perhaps I forgot to commit it?

**Comment by stephane-lb** [comment]
@ahundt yes you did. I will test your installation on my macbook, could you please open the bug trackers on your repo that we are able to track them there ? Thanks.

**Comment by stephane-lb** [comment]
@ahundt I tried to bypass it, unfortunately I got that : 
```
efusionConfig.cmake
```
So I suppose that you forgot a little bit more than just one file.

Don't know if you committed or not, but you did not push anything. Please let me know if you update your repo.

Thanks

**Comment by ahundt** [comment]
@stephane-lb I pushed the changes with the CMakeLists.txt Didn't see other missing files when I checked

**Pull request closed without merging**